### DISCUSSION TITLE
Script doesn't automatically load on @match ?

### DISCUSSION THREAD

**Discussion by reneverschoor** [violentmonkey/violentmonkey#1338] [Q&A] [answered]
I expect that a script is loaded when I open a page with an URL that matches `@match`, and unloads when I leave the page for a URL that doesn't match `@match`.

Test script:

```
// ==UserScript==
// @name          test_match
// @match         https://github.com/violentmonkey/violentmonkey/issues/*
// ==/UserScript==

'use strict';

(function () {
  console.log('Match!');
})();
```

Tested in Chrome and Vivaldi with Violentmonkey 2.13.0.
The above test script is the only script I've loaded.

I visit `https://github.com/violentmonkey/violentmonkey/issues`.
The Violentmonkey extension icon shows no scripts are loaded.
Refreshing the page makes no difference.
This is as expected, the URL doesn't match the script's `@match`.

I then click an issue, so the URL changes to `https://github.com/violentmonkey/violentmonkey/issues/nnnnn`.
I would expect that the script is now automatically loaded, but the Violentmonkey icon still shows no scripts are loaded.
Only after refreshing the page the script is loaded.
The counter changes to `1` and the message is logged in console.

I go back the the issues list page by clicking on `Issues` in the menu.
My expectation is that the script is now unloaded, as the URL no longer matches.
However, the Violentmonkey icon still shows `1`.
Only after a page refresh the counter shows that no scripts are loaded.

I hope I'm missing something basic :)

**Comment by tophf** [comment, marked as answer]
See https://github.com/violentmonkey/violentmonkey/issues/1028

**Comment by reneverschoor** [comment]
Ah, that makes perfect sense. 
Thanks!
